trio/_tests/test_exports.py

```diff
@@ -484,6 +496,16 @@ def lookup_symbol(symbol):
     assert not errors
 
 
+def test_nopublic_is_final() -> None:
+    """Check all NoPublicConstructor classes are also @final."""
+    assert class_is_final(_util.NoPublicConstructor)  # This is itself final.
+
+    for module in ALL_MODULES:
+        for name, class_ in module.__dict__.items():
+            if isinstance(class_, _util.NoPublicConstructor):
+                assert class_is_final(class_)
```

Comment:
Should we print out the class being tested so that when the test fails and stdout is printed, it's easier to check? (Or actually, `pytest` says what `class_` is through its `assert` debug stuff right?)

Reply:
It does, but I found that it was failing to include the full module name in some cases.